runtimes/runtime-blocking-bootstrap/src/main/java/com/flipkart/aesop/runtime/bootstrap/consumer/WaitPolicy.java

```diff
@@ -45,9 +45,8 @@ public void rejectedExecution(Runnable r, ThreadPoolExecutor e)
 	{
 		try
 		{
-			while (!e.isShutdown())
+			while (!e.isShutdown() && !e.getQueue().offer(r, _time, _timeUnit))
```

Comment:
Context - the idea of WaitPolicy is for a executor submission to "block" until it is accepted by the blocking queue.

The terminating conditions for the while loop should be either of
- executor is shutdown, or
- the runnable is successfully added to the queue

The bug was that previously, the while loop will go on until executor shutdown with every iteration re-adding the runnable to queue.